// eapp-uni/uni/ui.js
let isloading = false;
export default {
	previewImage(images,index=0){ 
		uni.previewImage({
					current:index, 
					urls:images 
				})
	}, 
	confirm(config, callbcak, cancel) {
		
		if(typeof config ==="string"){
			config = {
				content:config
			}
		}  
		
		uni.showModal( Object.assign( {
			title: "提示", 
		
			confirmText: "确定",
			cancelText: "取消",
			success:(res)=>{
			 if (res.confirm) { 
			            callbcak();
			        } else if (res.cancel && cancel) {
			            cancel();
			        }	
			},
			fail:cancel
		},config))
		
	},
	alert(message, callbcak) {

		// weui.alert(message);
		// return;
		uni.showModal({
			title: "提示",
			content: message+"",
			showCancel: false,
			confirmText: "确定",
			success:callbcak
		})

	},
	loading(msg = "请稍等") {
		isloading = true;
		uni.showToast({
			title: "加载中",
			icon: "loading",
			duration: 20000
		})
	},
	stopLoading() {
		if (isloading)
			this.stop_loading();
		isloading = false;
	},
	stop_loading() {
		uni.hideToast()
	},
	toast: {
		success(msg = "操作成功") {
			uni.showToast({
				title: msg
			})
		},
		text(msg) {
			uni.showToast({
				title: msg,
				position: "middle",
				icon: "none"
			})
		},
		fail(msg = "失败") {
			uni.showToast({
				title: msg,
				icon: "error"
			})
		}
	}

}
